src/components/SettingsPage/SettingsSyncCard.tsx

```diff
@@ -43,23 +52,146 @@ const SyncButton = ({ syncEntities }: SyncButtonProps) => {
   );
 };
 
+interface CancelSyncButtonProps {
+  cancelSync: () => Promise<void>;
+}
+
+const CancelSyncButton = ({ cancelSync }: CancelSyncButtonProps) => {
+  const [isCancelling, setIsCancelling] = useState(false);
+  return (
+    <IconButton
+      onClick={() => {
+        setIsCancelling(true);
+        cancelSync().finally(() => setIsCancelling(false));
+      }}
+      aria-busy={isCancelling}
+      aria-label="Cancel entity sync"
+    >
+      {isCancelling ? <CircularProgress /> : <CancelIcon />}
+    </IconButton>
+  );
+};
+
 export const SettingsSyncCard = () => {
   const catalogApi = useApi(catalogApiRef);
   const cortexApi = useApi(cortexApiRef);
   const extensionApi = useApi(extensionApiRef);
 
+  const [syncTaskProgressPercentage, setSyncTaskProgressPercentage] = useState<
+    number | null
+  >(null);
+
+  const [lastSyncedTime, setLastSyncedTime] = useState<string | null>(null);
+
   const syncEntities = async () => {
     const { items: entities } = await catalogApi.getEntities();
     const customMappings = await extensionApi.getCustomMappings?.();
     const groupOverrides = await extensionApi.getTeamOverrides?.(entities);
-    await cortexApi.syncEntities(entities, customMappings, groupOverrides);
+    const progress = await cortexApi.submitSyncTask(
+      entities,
+      customMappings,
+      groupOverrides,
+    );
+    setSyncTaskProgressPercentage(progress.percentage);
+  };
+
+  const cancelSync = async () => {
+    await cortexApi.cancelSync();
```

Comment:
esp with the exponential backoff, could take a while till the `PollingProgressBar` is unmounted -- what if we make this something like:
```
                await cortexApi.cancelSync();
                setSyncTaskProgressPercentage(null);
                setLastSyncedTime(
                  (await cortexApi.getLastSyncTime()).lastSynced,
                );
```